### PR TITLE
Make it rain!

### PR DESCRIPTION
**Pull request by tommbendall**
A storm is brewing in Gusto...

This pull requests adds some parametrizations to Gusto relating to rain: specifically involving the formation of rain from cloud water, the evaporation of rain water and a more sophisticated model of sedimentation.

And if your will to live hasn't already d-_rained_ away, it also updates `unsaturated_bubble.py` to be a warm rain test.

### CODE REVIEW

File: gusto/physics.py [hunk 1]
```diff
@@ -84,62 +97,285 @@ def __init__(self, state):
...
+            self.water_v.assign(self.water_v_new.interpolate())
+            self.water_c.assign(self.water_c_new.interpolate())
 
 
 class Fallout(Physics):
     """
     The fallout process of hydrometeors.
 
     :arg state :class: `.State.` object.
+    :arg moments: an integer denoting which rainfall scheme to use.
```
Comment (jshipton) [inline gusto/physics.py]: I was trying to think if there are more intuitive values for this argument but it's so complex that I'm not sure there is an obvious answer

Comment (wence-) [reply gusto/physics.py]: Use an Enum. 

AdvectedMoments.MEAN

And similar

Comment (thomasgibson) [reply gusto/physics.py]: https://docs.python.org/3/library/enum.html

Comment (tommbendall) [reply gusto/physics.py]: I don't think I understand enum well enough to see how to use it here!

Comment (wence-) [reply gusto/physics.py]: It just removes the need for magic numbers in the code.

```python
class AdvectedMoments(Enum):
   M0 = 0
   M3 = 1
   # any other moments here
```

```python
Fallout(..., moments=AdvectedMoments.M0, ...)
```

Now the value describes what is going on.

Comment (tommbendall) [reply gusto/physics.py]: I've tried to make this change now, possibly I still haven't exactly understood how to do it properly

**Comment by jshipton** [comment]
Looks great!

**Pull request merged**